Sources/Knit/Container+MainActor.swift

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright © Block, Inc. All rights reserved.
+//
+
+import Swinject
+
+extension Container {
+    // Register a service type's factory with the assumption that the registration
+    // will happen on the main thread.
+    //
+    // This method relies on the type eventually being resolved by a caller on the main
+    // thread using the knit generated resolver. If that call is not made on the main
+    // thread then a crash will occur.
+    @discardableResult
+    public func register<Service>(
```

Comment:
I've only added a single register function for now. I'll setup an ERB to generate them all in a future PR